src/main/php/PDepend/Source/AST/ASTNamespace.php

```diff
@@ -199,9 +197,8 @@ public function getTypes()
      * Adds the given type to this namespace and returns the input type instance.
      *
      * @param AbstractASTClassOrInterface $type
-     * @return AbstractASTClassOrInterface
      */
-    public function addType(AbstractASTType $type)
+    public function addType(AbstractASTType $type): AbstractASTClassOrInterface
```

Comment:
I think the return type should be AbstractASTType

```suggestion
    public function addType(AbstractASTType $type): AbstractASTType 
```

Reply:
It's actually the input that should be `AbstractASTClassOrInterface`. You can see this by it adding the type to $this->types which is a collection of AbstractASTClassOrInterface.

The type was to loose in 2.x and we overwrote it with the PHPDoc, this should be corrected in the native type, but I would like to wait till we are done with the automated cleanups first.

Reply:
I added it to https://github.com/pdepend/pdepend/pull/866